### PR TITLE
Add rule S3972 ('no-same-line-conditional') for TS/JS

### LINKED ISSUE #1511: Add rule S3972 ('no-same-line-conditional' in tslint) for TS/JS

### PR DESCRIPTION
**Pull request by christophe-zurn-sonarsource**
Fixes #1511

### CODE REVIEW

File: eslint-bridge/src/rules/no-same-line-conditional.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          }
+        });
+      },
+    };
+  },
+};
+
+function getSiblingIfStatements(node: estree.Node): SiblingIfStatement[] {
+  let statements: Array<estree.Node> = [];
+  if (node.type === "Program" || node.type === "BlockStatement") {
```
Comment (quentin-jaquier-sonarsource) [inline eslint-bridge/src/rules/no-same-line-conditional.ts]: I'm not sure matching on the 4 (line 43) of them and then checking types here is a nice way to do it.

File: eslint-bridge/src/rules/no-same-line-conditional.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            });
+          }
+        });
+      },
+    };
+  },
+};
+
+function getSiblingIfStatements(node: estree.Node): SiblingIfStatement[] {
+  let statements: Array<estree.Node> = [];
```
[the remark this answers is not in the document]
Comment (christophe-zurn-sonarsource) [reply eslint-bridge/src/rules/no-same-line-conditional.ts]: I followed @quentin-jaquier-sonarsource 's advice and match the statements separately, it makes it much cleaner

File: eslint-bridge/src/rules/no-same-line-conditional.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      {
+        // internal parameter for rules having secondary locations
+        enum: ["sonar-runtime"],
+      },
+    ],
+  },
+
+  create(context: Rule.RuleContext) {
+    return {
+      "Program, BlockStatement, FunctionBody, SwitchCase": function(node: estree.Node) {
```
[the remark this answers is not in the document]
Comment (christophe-zurn-sonarsource) [reply eslint-bridge/src/rules/no-same-line-conditional.ts]: Indeed, it does not exist in the library, I saw it on the github repo, but in estree it is `BlockStatement`

[the remark this answers is not in the document]
Comment (christophe-zurn-sonarsource) [reply eslint-bridge/src/rules/no-same-line-conditional.ts]: Nothing actually... It does not exist in the library, I saw it on the github repo, but in estree it is `BlockStatement`. Removed it

File: eslint-bridge/src/rules/no-same-line-conditional.ts (hunk 1 above)
Comment (christophe-zurn-sonarsource) [reply eslint-bridge/src/rules/no-same-line-conditional.ts]: Good idea indeed, it makes it much cleaner that way

**Pull request merged**